### DISCUSSION TITLE
Form based off Budibase Query Error

### DISCUSSION THREAD

**Discussion by JayP718** [Budibase/budibase#6197] [Help]
Hi!

We are trying to use a budibase query as the schema for a form but we keep receiving an internal server error. I'm unsure if maybe we are approaching this the wrong way. We've tried to use both a sql server view and a sql server table as the source data in the budibase query. The table has a primary key too. Seems like the error is a little vague as well and we checked the console and the error is just "500". The form works with the table listed as the schema.

Appreciate any advice. Thanks!

Jay Polanco

![Error Image](https://user-images.githubusercontent.com/42209052/172200799-1667bb54-c76e-4304-a6be-5c1567fa26de.png)

**Comment by melohagan** [comment]
Hey @JayP718 

Was there anymore information in the Network tab?

**Reply by JayP718** [reply]
@melohagan 

Not much else.

![network tab](https://user-images.githubusercontent.com/42209052/172213264-64a3054a-6127-4081-87f4-9f8ce8cc5b3f.png)

**Reply by melohagan** [reply]
Is data coming back within the **Data** tab? Is it just in the design that the 500 is occurring?

**Reply by JayP718** [reply]
Yes correct, the data is showing in the data tab. Just not working in the design tab

**Reply by JayP718** [reply]
And we are able to show the data in other type of controls such as the card.

**Reply by melohagan** [reply]
The URL is `http://nysap:10000/...` What is *nysap* ? 

If you're self-hosted, you could share any error logs from the **bbapps** and **bbworker** services.

**Reply by JayP718** [reply]
Thanks for the quick response @melohagan! So this is what is appearing in the logs for bbapps 

```
Trace: Datasource is not configured fully.
    at /app/dist/api/index.js:67:21
    at Generator.throw (<anonymous>)
    at rejected (/app/dist/api/index.js:6:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
ERROR [1654548217790] (28 on f7fc12e22927): Datasource is not configured fully.
```

```
 "type": "Error",
      "message": "failed with status code 500",
      "stack":
          Error: failed with status code 500
              at ServerResponse.onResFinished (/app/node_modules/pino-http/logger.js:77:38)
              at ServerResponse.emit (events.js:412:35)
              at ServerResponse.emit (domain.js:475:12)
              at ServerResponse.emitted (/app/node_modules/emitter-listener/listener.js:134:21)
              at onFinish (_http_outgoing.js:792:10)
              at callback (internal/streams/writable.js:513:21)
              at afterWrite (internal/streams/writable.js:466:5)
              at afterWriteTick (internal/streams/writable.js:453:10)
              at processTicksAndRejections (internal/process/task_queues.js:81:21)
    }
```

Nothing seems to pop out in bbworker logs.

**Reply by melohagan** [reply]
Have you tried removing the datasource, and then re-adding and re-fetching? Maybe there was a hiccup in the initial setup.

**Reply by QZTIAN** [reply]
We update Budibase to the most recent version and now the error messages disappears. However, when I use the query for Form Schema, I cannot get any fields to select.

**Reply by melohagan** [reply]
Hey @QZTIAN 
Do you mean there are no field options showing in the name dropdown? e.g. nothing showing here?
<img width="217" alt="Screenshot 2022-06-07 at 21 22 54" src="https://user-images.githubusercontent.com/101575380/172475382-f3a389d4-a3c4-4752-91e3-26c28f270a11.png">

**Reply by QZTIAN** [reply]
You are right! There are no options in the dropdown. Thanks!!

**Reply by melohagan** [reply]
Could you share some screenshots please? 

- your query including the output schema (in the data tab)
- your component tree (in the design tab)
- the settings panel of your Form

**Reply by QZTIAN** [reply]
Sure!
![designtree](https://user-images.githubusercontent.com/107056035/172479419-39b9b285-ede7-4aad-a721-3bb9f361a507.jpg)
![Screenshot 2022-06-07 164348](https://user-images.githubusercontent.com/107056035/172479422-3be324ae-abb6-472d-ad4a-daae226477c7.jpg)
![image](https://user-images.githubusercontent.com/107056035/172479642-0d19b5ac-f46d-46fa-8f87-5f2cdf07bfef.png)
Sure!

**Reply by melohagan** [reply]
@QZTIAN 
I'm wondering what the data types are, if you click on the **Schema** tab and show me that.
Does the summons number appear if you use a **Number Field** instead of a **Text Field** for example?

**Reply by QZTIAN** [reply]
![image](https://user-images.githubusercontent.com/107056035/172642386-10b00f8d-0892-4179-8057-5b9aee5f0c69.png)
Here is the screenshot of the Schema.

**Reply by melohagan** [reply]
Thanks @QZTIAN I've been able to replicate the bug. I've raised an issue here: https://github.com/Budibase/budibase/issues/6238
